lib/uwb/UwbMacAddress.cxx

```diff
@@ -25,6 +27,43 @@ UwbMacAddress::UwbMacAddress(const UwbMacAddress& other) :
     InitializeView();
 }
 
+UwbMacAddress::UwbMacAddress(std::string addressString, UwbMacAddressType addressType)
+{
+    std::regex shortRegex("^([0-9A-Fa-f]{2}:){1}([0-9A-Fa-f]{2})$");
+    std::regex extendedRegex("^(([0-9A-Fa-f]{2}:){7}[0-9A-Fa-f]{2})$");
+    assert(std::regex_match(addressString, shortRegex) || std::regex_match(addressString, extendedRegex));
+
+    std::array<uint8_t, ShortLength> shortAddress{};
+    std::array<uint8_t, ExtendedLength> extendedAddress{};
+    try {
+        std::stringstream ss(addressString);
+        for (auto i = 0; i < (addressType == UwbMacAddressType::Short ? ShortLength : ExtendedLength); i++) {
+            if (ss) {
+                std::string addressByte;
+                getline(ss, addressByte, ':');
+                const auto byteValue = static_cast<uint8_t>(std::stoi(addressByte));
+                if (addressType == UwbMacAddressType::Short) {
+                    shortAddress[i] = byteValue;
+                } else {
+                    extendedAddress[i] = byteValue;
+                }
+            } else {
+                throw std::runtime_error("invalid mac address string");
+            }
+        }
+
+        if (addressType == UwbMacAddressType::Short) {
+            *this = UwbMacAddress(shortAddress);
+        } else {
+            *this = UwbMacAddress(extendedAddress);
+        }
+    }
+    catch (const std::exception& e)
+    {
+        throw std::invalid_argument("Invalid mac address string: " + addressString);
```

Comment:
We're trying to avoid throwing in this class. You can avoid that if you make this constructor private and then wrap its use in a static factory function which returns a `std::optional<::uwb::UwbMacAddress>` instead, eg.

```
static
std::optional<::uwb::UwbMacAddress> FromString(const std::string addressString, UwbMacAddressType macAddressType)
{
    try {
        return UwbMacAddress(addressString, macAddressType);
    } catch (...) {
        return std::nullopt;
    }
}
```

This way the caller doesn't have to wrap creating a mac address in a try/catch block.

Reply:
I made this change along with some others to make all error handling more graceful. It's much better now!